src/Data/Dwarf.hs

```diff
@@ -430,92 +423,171 @@ getDieAndSiblings cuContext = do
-    DW_FORM_udata -> DW_ATVAL_UINT <$> getULEB128
-    DW_FORM_ref_addr -> DW_ATVAL_UINT <$> desrGetOffset end enc
-    DW_FORM_ref1 -> DW_ATVAL_REF . inCU cu <$> getWord8
-    DW_FORM_ref2 -> DW_ATVAL_REF . inCU cu <$> derGetW16 end
-    DW_FORM_ref4 -> DW_ATVAL_REF . inCU cu <$> derGetW32 end
-    DW_FORM_ref8 -> DW_ATVAL_REF . inCU cu <$> derGetW64 end
-    DW_FORM_ref_udata -> DW_ATVAL_REF . inCU cu <$> getULEB128
+    DW_FORM_strx -> parseStrxOfBytesz getULEB128
+    DW_FORM_addrx -> parseAddrxOfBytesz getULEB128
+    DW_FORM_strx1 -> parseStrxOfBytesz getWord8
+    DW_FORM_strx2 -> parseStrxOfBytesz (derGetW16 end)
+    DW_FORM_strx3 -> parseStrxOfBytesz (derGetW24 end)
+    DW_FORM_strx4 -> parseStrxOfBytesz (derGetW32 end)
+    DW_FORM_addrx1 -> parseAddrxOfBytesz getWord8
+    DW_FORM_addrx2 -> parseAddrxOfBytesz (derGetW16 end)
+    DW_FORM_addrx3 -> parseAddrxOfBytesz (derGetW24 end)
+    DW_FORM_addrx4 -> parseAddrxOfBytesz (derGetW32 end)
+    DW_FORM_ref1 -> RealizedAttr . DW_ATVAL_REF . inCU cu <$> getWord8
+    DW_FORM_ref2 -> RealizedAttr . DW_ATVAL_REF . inCU cu <$> derGetW16 end
+    DW_FORM_ref4 -> RealizedAttr . DW_ATVAL_REF . inCU cu <$> derGetW32 end
+    DW_FORM_ref8 -> RealizedAttr . DW_ATVAL_REF . inCU cu <$> derGetW64 end
+    DW_FORM_ref_udata ->  RealizedAttr . DW_ATVAL_REF . inCU cu <$> getULEB128
     DW_FORM_indirect -> getForm cuContext . DW_FORM =<< getULEB128
-    -- new in Dwarf version 4
-    DW_FORM_sec_offset -> DW_ATVAL_UINT <$> desrGetOffset end enc
-    DW_FORM_exprloc -> DW_ATVAL_BLOB <$> getByteStringLen getULEB128
-    DW_FORM_flag_present -> pure $ DW_ATVAL_BOOL True
-    DW_FORM_ref_sig8 -> DW_ATVAL_UINT . fromIntegral <$> derGetW64 end
-    DW_FORM_strx -> unimplForm form
-    DW_FORM_addrx -> unimplForm form
-    DW_FORM_ref_sup4 -> unimplForm form
-    DW_FORM_strp_sup -> unimplForm form
-    DW_FORM_data16 -> unimplForm form
-    DW_FORM_line_strp -> unimplForm form
-    DW_FORM_implicit_const -> unimplForm form
-    DW_FORM_loclistx -> unimplForm form
-    DW_FORM_rnglistx -> undefined
-    DW_FORM_ref_sup8 -> undefined
-    DW_FORM_strx1 -> unimplForm form
-    DW_FORM_strx2 -> unimplForm form
-    DW_FORM_strx3 -> unimplForm form
-    DW_FORM_strx4 -> unimplForm form
-    DW_FORM_addrx1 -> unimplForm form
-    DW_FORM_addrx2 -> unimplForm form
-    DW_FORM_addrx3 -> unimplForm form
-    DW_FORM_addrx4 -> unimplForm form
-    _ -> unimplForm form
+    _ -> RealizedAttr <$> getEvaluableForm dc end enc tgt form
+  where
+      -- shared structure between addrx and strx operations where an offset is used to index an array and that value is an offset into a further table
+      -- takes a parser for the offset, a function that consumes the offset and produces the final attribute values and a way to retrieve the base offset
+      -- for the target table
+      parseIntegralOff :: 
+        (MonadFail m, Show a, Integral a) =>
+        Sections ->
+        Get a ->
```

Comment:
Feel free to ignore this comment. It seems slightly odd to pass a `Get a` only to immediately run it. Callers could instead call `parseIntegralOff ... =<< getter` if the arguments were in a different order.